sympy/solvers/solveset.py

```diff
@@ -239,6 +239,22 @@ def _invert_real(f, g_ys, symbol):
         if g is not S.One:
             return _invert_real(h, imageset(Lambda(n, n/g), g_ys), symbol)
 
+        if g_ys == FiniteSet(S.Zero):
+            res = S.EmptySet
+            test = True
```

Comment:
You may use [for-else](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) to get rid of `test` variable.

Reply:
Sure! Thanks! I will change the code accordingly.